include/ada/mimesniff/mimetype.h

```diff
@@ -39,27 +78,29 @@ struct mimetype {
   std::string serialized() const noexcept {
     std::string base = essence();
 
-    for (const auto &i : parameters) {
-      base += ";";
-      base += std::string(i.first) + "=";
-
-      if (i.second.empty() || !contains_only_http_tokens(i.second)) {
+    for (const key_value &i : parameters) {
+      base += ';';
+      base += std::string(i.key()) + "=";
+      if (i.value().empty() || !contains_only_http_tokens(i.value())) {
         // Precede each occurrence of U+0022 (") or U+005C (\) in value with
         // U+005C (\).
-        std::string value = i.second;
-        auto ptr = value.find_first_of("\"\\");
-
-        while (ptr != std::string_view::npos) {
-          value.insert(ptr, "\\");
-          ptr = value.find_first_of("\"\\", ptr);
+        base += '\"';
+        for(char c : i.value()) {
```

Comment:
What is the reason for going with a character by character loop, rather than the previous implementation?

Reply:
Honestly, the previous version fell into an infinite loop, and so I just wanted to simplify it.

Later, we should try different implementations and pick the fastest one.

My goal was solely to get the tests to pass.